lib/stat_tracker.rb

```diff
@@ -90,4 +91,130 @@ def average_goals_by_season
       average_of_goals_in_a_season(season.first.season)
     end
   end
+
+  def count_of_teams
+    @teams.length
+  end
+
+  def average_goals_by_team(team_id, hoa = nil)
+    goals = 0
+    games = 0
+    @game_teams.each do |game_team|
+      if hoa != nil
+        if game_team.team_id == team_id && game_team.hoa == hoa
+          goals += game_team.goals
+          games += 1
+        end
+      elsif game_team.team_id == team_id
+          goals += game_team.goals
+          games += 1
+      end
+    end
+    return 0 if games == 0
+    (goals.to_f / games.to_f).round(2)
+  end
+
+  def unique_team_ids
+    @game_teams.uniq { |game_team| game_team.team_id }.map{ |game_team| game_team.team_id }
+  end
+
+  def team_by_id(team_id)
+    @teams.find { |team| team.team_id == team_id }
+  end
+
+  def best_offense
+    id = unique_team_ids.max_by {|team_id| average_goals_by_team(team_id)}
+    team_by_id(id).team_name
+  end
+
+  def worst_offense
+    id = unique_team_ids.min_by {|team_id| average_goals_by_team(team_id)}
+    team_by_id(id).team_name
+  end
+
+  def highest_scoring_visitor
+    id = unique_team_ids.max_by {|team_id| average_goals_by_team(team_id, "away")}
+    team_by_id(id).team_name
+  end
+
+  def highest_scoring_home_team
+    id = unique_team_ids.max_by {|team_id| average_goals_by_team(team_id, "home")}
+    team_by_id(id).team_name
+  end
+
+  def lowest_scoring_visitor
+    id = unique_team_ids.min_by {|team_id| average_goals_by_team(team_id, "away")}
+    team_by_id(id).team_name
+  end
+
+  def lowest_scoring_home_team
+    id = unique_team_ids.min_by {|team_id| average_goals_by_team(team_id, "home")}
+    team_by_id(id).team_name
+  end
+
+  def most_accurate_team(season)
+    season_games = @games.find_all{|game| game.season == season}
+    season_game_ids = season_games.map{|game| game.game_id}
+    team_performances = @game_teams.find_all{|team| season_game_ids.include?(team.game_id)}
```

Comment:
this part definitely needs to be a method on its own to be called by all these season methods